runtime/runtime/src/lib.rs

```diff
@@ -1117,13 +1125,77 @@ impl Runtime {
             get(&state_update, &TrieKey::DelayedReceiptIndices)?.unwrap_or_default();
         let initial_delayed_receipt_indices = delayed_receipts_indices.clone();
 
+        let mut machine =
+            WasmMachine::new(apply_state.config.wasm_config.clone(), apply_state.cache.clone());
+
+        let prefetch = false;
+        if prefetch {
+            let mut acc = Vec::new();
+            let mut prefetch_receipt = |receipt: &Receipt| {
+                match &receipt.receipt {
+                    ReceiptEnum::Data(_) => {
+                        // TODO(matklad): fetch postponed receipt
+                    }
+                    ReceiptEnum::Action(action_receipt) => {
+                        for (action_index, action) in action_receipt.actions.iter().enumerate() {
+                            if let Action::FunctionCall(_) = action {
+                                if let Ok(Some(account)) =
+                                    get_account(&state_update, &receipt.receiver_id)
+                                {
+                                    let code = crate::cache::get_code(account.code_hash, || {
+                                        near_store::get_code(
+                                            &state_update,
+                                            &receipt.receiver_id,
+                                            Some(account.code_hash),
+                                        )
+                                    });
+
+                                    if let Ok(Some(code)) = code {
+                                        let action_hash = create_action_hash(
+                                            apply_state.current_protocol_version,
+                                            &receipt,
+                                            &apply_state.prev_block_hash,
+                                            &apply_state.block_hash,
+                                            action_index,
+                                        );
+                                        acc.push((action_hash, code));
```

Comment:
We want to preload not a particular contract per-se, but a specific contract methond execution, as that might also require additional resources. IIUC, wasmer 0.17 at least wants to allocate some fresh virtual memory for each call.

I originally actually went with using code hash, but @olonho convinced me that keying by actions is a better approach.

Reply:
Why is loading specific contract method execution more preferred? From my perspective this is not great because you recompute action_hash here, which isn't free. It also makes the code more cumbersome

Reply:
It's cheap recomputing, as input size is rather small, and we shall identify the particular invocation, not just contract, to make it more flexible (i.e. if we'll want to resolve *which* method to be invoked in preloader as well, or otherwise optimize execution in context of particular method to be invoked).